### DISCUSSION TITLE
Max height being asked for private ways

### DISCUSSION THREAD

**Discussion by arrival-spring** [streetcomplete/StreetComplete#3879] [Q&A]
Is there a reason that `AddMaxHeight` does not have `access !~ private|no` for any highway other than service?

https://github.com/streetcomplete/StreetComplete/blob/5ffc156c2f5ac826a7e60909914caa68ed751321/app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeight.kt#L27-L34

It looks to be written like that intentionally

Looking at git blame led me to https://github.com/streetcomplete/StreetComplete/pull/1516#issuecomment-521405336 where the query is written like that in overpass code.

I feel like it should be something like
`highway ~ motorway|motorway_link|trunk|trunk_link|primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential|living_street|service|track|road`
`and access !~ private|no and vehicle !~ private|no and motor_vehicle !~ private|no`

(Triggered by spotting: https://www.openstreetmap.org/note/2746280
Which is asking for `highway=track` with `access=no`)

**Comment by riQQ** [comment]
See also #3026

**Reply by arrival-spring** [reply]
I feel like `access~private|no` is rare on things unclassified and above, but I would at the very least suggest adding `track` to private ways that shouldn't be asked for

**Reply by matkoniecz** [reply]
I would expect that in such cases it would be still generally visible or accessible...

Not sure, but given example appears to be such case where survey was still possible.

(though I would probably merge tested PR adding such filter)
